src/profile/components/profile/Profile.tsx

```diff
@@ -93,60 +39,64 @@ function Profile(): React.ReactElement {
         return 'appName';
     }
   };
-  if (failedToFetchUserProfileData) {
-    return (
-      <PageLayout title={'notification.defaultErrorTitle'}>
-        <div
-          className={styles['error-wrapper']}
-          data-testid="profile-check-error-layout"
-        >
-          <div className={responsive['max-width-centered']}>
-            <Notification
-              type={'error'}
-              label={t('notification.defaultErrorTitle')}
-            >
-              {t('profile.loadErrorText')}
-            </Notification>
-            <div className={styles['error-button-wrapper']}>
-              <Button
-                onClick={() => checkProfileExists()}
-                data-testid={'profile-check-error-reload-button'}
-              >
-                {t('profile.reload')}
-              </Button>
-              <Button
-                onClick={() => authService.logout()}
-                data-testid={'profile-check-error-logout-button'}
-                variant={'secondary'}
+
+  if (isProfileLoadComplete()) {
+    if (didProfileLoadFail()) {
+      return (
+        <PageLayout title={'notification.defaultErrorTitle'}>
+          <div
+            className={styles['error-wrapper']}
+            data-testid="profile-check-error-layout"
+          >
+            <div className={responsive['max-width-centered']}>
+              <Notification
+                type={'error'}
+                label={t('notification.defaultErrorTitle')}
               >
-                {t('nav.signout')}
-              </Button>
+                {t('profile.loadErrorText')}
+              </Notification>
+              <div className={styles['error-button-wrapper']}>
+                <Button
+                  onClick={() => reloadProfile()}
+                  data-testid={'profile-check-error-reload-button'}
+                >
+                  {t('profile.reload')}
+                </Button>
+                <Button
+                  onClick={() => authService.logout()}
+                  data-testid={'profile-check-error-logout-button'}
+                  variant={'secondary'}
+                >
+                  {t('nav.signout')}
+                </Button>
+              </div>
             </div>
           </div>
-        </div>
+        </PageLayout>
+      );
+    }
+    if (hasExistingProfile()) {
+      return (
+        <PageLayout title={getPageTitle()}>
+          <ViewProfile />
+        </PageLayout>
+      );
+    }
+    return (
+      <PageLayout title={getPageTitle()}>
+        <CreateProfile
+          tunnistamoUser={props.user}
+          onProfileCreated={() => reloadProfile()}
+        />
+        ;
       </PageLayout>
     );
   }
 
   return (
-    <PageLayout title={getPageTitle()} disableFocusing>
```

Comment:
Should the `disableFocusing` property still be present somewhere in this `Profile` component?

Reply:
True. Added to loader in WithAuthCheck too